android/src/main/java/com/reactnativeshopify/RNShopifyModule.java

```diff
@@ -0,0 +1,354 @@
+
+package com.reactnativeshopify;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.Iterator;
+
+import org.json.*;
+
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReadableArray;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.ReadableMapKeySetIterator;
+import com.facebook.react.bridge.ReadableType;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.bridge.WritableNativeMap;
+import com.facebook.react.bridge.WritableArray;
+import com.facebook.react.bridge.WritableNativeArray;
+import com.facebook.react.bridge.Arguments;
+
+import com.shopify.buy.dataprovider.*;
+import com.shopify.buy.model.*;
+
+public class RNShopifyModule extends ReactContextBaseJavaModule {
+
+  private final ReactApplicationContext reactContext;
+  private BuyClient buyClient;
+
+  public RNShopifyModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+    this.reactContext = reactContext;
+  }
+
+  @Override
+  public String getName() {
+    return "RNShopify";
+  }
+
+  @ReactMethod
+  public void initialize(String domain, String key) {
+
+    //Application ID is always 8, as stated in official documentation from Shopify
+    buyClient = new BuyClientBuilder()
+    .shopDomain(domain)
+    .apiKey(key)
+    .appId("8")
+    .applicationName("com.reactnativeshopify")
```

Comment:
The package name of our application is different than that`(com.shoutemapp`). I think that too needs to be configurable in the long run. It is possible it really doesn't matter but it would be good to check if there would be any name clashes if we had two applications using this extension.